### PR TITLE
Refactor scripts

### PR DESCRIPTION
**Pull request by codebuster22**
1) Now using getSigners() in scripts.
2) Added single line comment for each scripts
3) Updated the docs to send transaction only after receipt received.
4) Added require statement to check if the transaction hash is already signer or not at Signer.sol

### CODE REVIEW

File: docs/contract-deployment.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Deploying contract
+
+## Deploy Contract
```
Comment (kinspotting) [inline docs/contract-deployment.md:3]: `Deploy Contracts`

File: docs/contract-deployment.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+to deploy ```SeedFactory```, ```Seed```, ```Signer``` and set mastercopy at ```SeedFactory```
+```bash
```
Comment (kinspotting) [inline docs/contract-deployment.md:6]: Deploy on rinkeby:

File: docs/contract-deployment.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+npm run deploy:contracts:rinkeby
+```
+
```
Comment (kinspotting) [inline docs/contract-deployment.md:9]: Deploy on mainnet:
```
npm run deploy:contracts:mainnet
```

File: docs/contract-deployment.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+## Change Ownership
```
Comment (kinspotting) [inline docs/contract-deployment.md:10]: pls also add mainnet

**Review by kinspotting — CHANGES_REQUESTED** [review]
Documentation doesn't represent the changes in scripts

**Comment by fabianschu** [comment]
@kinspotting Just as a side note: the README is updated in https://github.com/PrimeDAO/contracts-v2/pull/56, where we change the deployment logic more generally.

File: package.json [hunk 5]
```diff
@@ -4,6 +4,10 @@
   "description": "new experimental version of PrimeDAO contracts",
   "main": "index.js",
   "scripts": {
+    "deploy:contracts:mainnet": "npx hardhat run scripts/deploy_contracts.js --network mainnet",
+    "change:factoryOwner:mainnet": "npx hardhat run scripts/change_owner.js --network mainnet",
+    "add:delegate:mainnet": "npx hardhat run scripts/add_delegate.js --network mainnet",
```
Comment (kinspotting) [inline package.json:9]: there is no such script in the `scripts` folder

File: package.json [hunk 6]
```diff
@@ -4,6 +4,10 @@
...
+    "send:safeTrx:mainnet": "npx hardhat run scripts/add_tx_to_safe.js --network mainnet",
```
Comment (kinspotting) [inline package.json:10]: same

File: package.json [hunk 7]
```diff
@@ -4,6 +4,10 @@
...
     "deploy:contracts:rinkeby": "npx hardhat run scripts/deploy_contracts.js --network rinkeby",
```
Comment (kinspotting) [inline package.json:11]: same

File: package.json [hunk 8]
```diff
@@ -4,6 +4,10 @@
...
     "change:factoryOwner:rinkeby": "npx hardhat run scripts/change_owner.js --network rinkeby",
     "add:delegate:rinkeby": "npx hardhat run scripts/add_delegate.js --network rinkeby",
```
Comment (kinspotting) [inline package.json:13]: pls adjust `npm`commands accordingly, also the scripts in `mainnet` and `rinkeby` folders are named in camel case and in the `scripts` folder the `_` is used, pls choose one

File: docs/contract-deployment.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+### For mainnet
+
+to deploy ```SeedFactory```, ```Seed```, ```Signer``` and set mastercopy at ```SeedFactory```
+```bash
+npm run deploy:contracts:mainnet
+```
+
+to transferOwnership to Safe
+```bash
+npm run change:factoryOwner:mainnet
```
Comment (kinspotting) [inline docs/contract-deployment.md:14]: not a clear command imho

File: docs/contract-deployment.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+to add Signer contract as delegate for Safe
+```bash
+npm run add:delegate:mainnet
```
Comment (kinspotting) [inline docs/contract-deployment.md:19]: same here

**Review by kinspotting — CHANGES_REQUESTED** [review]

**Review by kinspotting — APPROVED** [review]

**Pull request merged**